File: LintCode/366. Fibonacci.py
# Your submission beats 96.60% Submissions!
class Solution:
    """
    @param n: an integer
    @return: an ineger f(n)
    """
    def fibonacci(self, n):
        fib = [0, 1]
        while len(fib)<n:
            fib.append(fib[-1]+fib[-2])
        return fib[n-1]

File: LintCode/366. Fibonacci.cpp
// Your submission beats 66.80% Submissions!
#include <iostream>
using namespace std;

class Solution
{
public:
    /**
     * @param n: an integer
     * @return: an ineger f(n)
     */
    int fibonacci(int n)
    {
        int fib[n];
        fib[0] = 0, fib[1] = 1;
        for (int i = 2; i < n; i++)
        {
            fib[i] = fib[i - 1] + fib[i - 2];
        }
        return fib[n - 1];
    }
};